CONTRIBUTING.md

```diff
@@ -97,137 +76,310 @@ script before each commit and abort the commit on error. To
+- Now start using the extension to test your changes.
+
+- For updates, rebuild and package your extension and publish an update from
+the Azure DevOps marketplace. That will automatically update the extension in
+your project.
+
+- For the real time live syncing to work, our service needs to know your
+publisher id and your extension's unique key. To enable real time updates for
+your test extension, please
+[reach out to us](https://github.com/microsoft/vsts-extension-retrospectives/issues)
+with your publisher id and the
+[unique key](https://docs.microsoft.com/en-us/azure/devops/extend/develop/auth?view=vsts#get-your-extensions-key)
+of your extension.
+
+### Test with Hot Reload and Debug
+
+Test changes by loading changes locally without having to re-package and
+re-publish the extension in the marketplace.
+
+---
+
+#### Hot Reload Prerequisites
+
+1. [Azure DevOps Extension Hot Reload and Debug](https://github.com/microsoft/azure-devops-extension-hot-reload-and-debug)
+
+2. [Visual Studio Code](https://code.visualstudio.com/download)
+
+3. [Firefox](https://www.mozilla.org/en-US/firefox/)
+
+4. [Debugger for Firefox](https://marketplace.visualstudio.com/items?itemName=firefox-devtools.vscode-firefox-debug) VS Code extension
+
+---
+
+1. In the 'RetrospectiveExtension.Frontend' folder, create the
+'vss-extension-dev.json' file using the template file
+`vss-extension-dev.json.template` for reference.
+2. Update the 'webpack.config.js' to enable source maps. Set the devtool
+property to `inline-source-map`. Also set devServer.https to true and
+devServer.port to 3000.
+
+    ```js
+    module.exports = {
+      devtool: 'inline-source-map',
+      devServer: {
+        https: true,
+        port: 3000,
+        static: {
+          directory: path.join(__dirname),
+        }
+      },
+    ...
+    ```
+
+3. Set `output.publicPath` to `/dist/` in the webpack.config.json file. This
+will allow webpack to serve files from `https://localhost:3000/dist`.
+
+    ```js
+    module.exports = {
+        output: {
+          publicPath: "/dist/"
+          // ...
+        }
+        // ..
+    };
+    ```
+
+4. In the root of the project, create a folder named `.vscode`. In there,
+create a file named `launch.json`, which will help to set up a debug
+configuration for VS Code that launches Firefox with the correct path mappings.
+Inside of this file, you will add a path mapping with `url` set to
+`webpack:///` and have the path set to
+`${workspaceFolder}/RetrospectiveExtension.Frontend/`. Also set the reAttach
+property on the configuration to true to avoid restarting Firefox every time
+you debug.
+
+    ```json
+    {
+      "version": "0.2.0",
+      "configurations": [
+        {
+          "name": "Launch Firefox",
+          "type": "firefox",
+          "request": "launch",
+          "url": "https://localhost:3000/",
+          "reAttach": true,
+          "pathMappings": [
+                {
```

Comment:
nit: It looks like the indentation is not correct here probably due to the "Add a suggestion" syntax.

Reply:
ah you are correct - I've update the indentation